runtimes/eoapi/stac/eoapi/stac/extensions.py

```diff
@@ -104,3 +109,126 @@ async def stac_viewer(
             return RedirectResponse(url)
 
         app.include_router(self.router, tags=["TiTiler Extension"])
+
+
+@attr.s
+class HTMLorJSONGetRequest(APIRequest):
+    """HTML or JSON output."""
+
+    f: Annotated[
+        Optional[Literal["json", "html"]],
+        Query(description="Response MediaType."),
+    ] = attr.ib(default=None)
+
+
+@attr.s
+class HTMLorGeoGetRequest(APIRequest):
+    """HTML or GeoJSON output."""
+
+    f: Annotated[
+        Optional[Literal["geojson", "html"]],
+        Query(description="Response MediaType."),
+    ] = attr.ib(default=None)
+
+
+@attr.s(kw_only=True)
+class HTMLorJSONOutputExtension(ApiExtension):
+    """TiTiler extension."""
+
+    GET = HTMLorJSONGetRequest
+    POST = None
+
+    def register(self, app: FastAPI) -> None:
+        pass
+
+
+@attr.s(kw_only=True)
+class HTMLorGeoOutputExtension(ApiExtension):
+    """TiTiler extension."""
+
+    GET = HTMLorGeoGetRequest
+    POST = None
+
+    def register(self, app: FastAPI) -> None:
+        pass
+
+
+@attr.s(kw_only=True)
+class HTMLorSchemaGetRequest(APIRequest):
+    f: Annotated[
+        Optional[Literal["jsonschema", "html"]],
+        Query(description="Response MediaType."),
+    ] = attr.ib(default=None)
+
+
+@attr.s(kw_only=True)
+class CollectionFilterGetRequestModel(CollectionUri, HTMLorSchemaGetRequest):
+    pass
+
+
+@attr.s
+class SearchFilterExtension(core.SearchFilterExtension):
+    """Item Search Filter Extension."""
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        self.router.prefix = app.state.router_prefix
+        self.router.add_api_route(
+            name="Queryables",
+            path="/queryables",
+            methods=["GET"],
+            responses={
+                200: {
+                    "content": {
+                        MimeTypes.jsonschema.value: {},
+                        MimeTypes.html.value: {},
+                    },
+                },
+            },
+            response_class=self.response_class,
+            endpoint=create_async_endpoint(
+                self.client.get_queryables, HTMLorSchemaGetRequest
+            ),
+        )
+        app.include_router(self.router, tags=["Filter Extension"])
```

Comment:
we need custom Filter extension client because we need to configure output models